File: abc/abc024/a.cpp
// https://atcoder.jp/contests/abc024/submissions/26617587

#include <bits/stdc++.h>
using namespace std;
int main() {
  int a, b, c, k, s, t; cin>>a>>b>>c>>k>>s>>t;
  if (s+t>=k) {a-=c; b-=c;}
  cout << a*s+b*t << endl;
}

File: abc/abc024/a.py
# https://atcoder.jp/contests/abc024/submissions/38180771

a, b, c, k = map(int, input().split())
s, t = map(int, input().split())

ans = s * a + t * b
if (s + t) >= k:
  ans -= (s + t) * c

print(ans)
